configure.py

```diff
@@ -2554,7 +2554,10 @@ def configure_using_cmake(args):
         settings['Scylla_CLANG_INLINE_THRESHOLD'] = args.clang_inline_threshold
 
     source_dir = os.path.realpath(os.path.dirname(__file__))
-    build_dir = os.path.join(source_dir, 'build')
+    if os.path.isabs(outdir):
```

Comment:
nit, instead of referencing a global variable, I recommend using `args.build_dir`. This approach offers several benefits:

By directly using the command-line argument, we make it explicit that the build directory is a user-specified option. This change makes the code's intent clearer and ties the configuration directly to the user's input. Eliminating global variable references can help prevent potential side effects and make the code more predictable.

Reply:
I wholly agree with you, but ultimately I opted for `outdir` for consistency's sake. Pre-patch, the only mention of `args.build_dir` is when it is snatched into the `outdir` global, soon after `args = arg_parser.parse_args()`; and then the entire non-CMake logic in `configure.py` references `outdir` (without ever modifying it, AFAICT). I didn't want to diverge from that usage. (Obviously, if I had written v1 of this patch with `args.build_dir`, someone would have immediately commented, "you're being inconsistent with current `outdir` usage" 😉)

Historically (git-blame and git-log help):
1. Commit 4f0f97773e0d introduced `outdir` merely as a symbolic name for the string `build`.
2. In commit be3776ec2ab1, we introduced the `--build-dir` option; the idea was `outdir = os.path.abspath(args.build_dir)`.
3. That turned out to be a bad idea, so it was short-circuited in commit af0c0ee8af48, as `outdir = args.build_dir`.

You could argue therefore that the `outdir = args.build_dir` assignment we have today is technical debt from commit af0c0ee8af48. At that point, no reason _remained_ for maintaining an `outdir` global separately from `args.build_dir`. In other words, instead of assigning `args.build_dir` to `outdir`, all `outdir` references should have been replaced with `args.build_dir`.

Do you want me to clean that up now, as a pre-requisite for my patch?

Because, in that case, it would be logical and consistent, for my own patch as well, to use `args.build_dir`.

Reply:
> I didn't want to diverge from that usage.

i don't insist. as this change is an improvement per-se.

> (Obviously, if I had written v1 of this patch with `args.build_dir`, someone would have immediately commented, "you're being inconsistent with current `outdir` usage" 😉)

i will send a follow-up change to implement my proposal. because the "inconsistency" is expected. i strive to make the new code more maintainable even at the expense of being inconsistent with the existing code.

> 
> You could argue therefore that the `outdir = args.build_dir` assignment we have today is technical debt from commit [af0c0ee](https://github.com/scylladb/scylladb/commit/af0c0ee8af48f1691a8f8a379a442da5eeeb5f8a). At that point, no reason _remained_ for maintaining an `outdir` global separately from `args.build_dir`. In other words, instead of assigning `args.build_dir` to `outdir`, all `outdir` references should have been replaced with `args.build_dir`.
> 
> Do you want me to clean that up now, as a pre-requisite for my patch?
> 

i don't think it is a pre-requisite. as i think the non-cmake code will go away very soon.

> Because, in that case, it would be logical and consistent, for my own patch as well, to use `args.build_dir`.

Reply:
> > Do you want me to clean that up now, as a pre-requisite for my patch?
> 
> i don't think it is a pre-requisite. as i think the non-cmake code will go away very soon.

OK, let me resubmit with just `args.build_dir`.

Reply:
Done in v2.